app/src/main/java/com/iron/espresso/presentation/home/mystudy/studydetail/ModifyStudyViewModel.kt

```diff
@@ -0,0 +1,119 @@
+package com.iron.espresso.presentation.home.mystudy.studydetail
+
+import android.net.Uri
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.iron.espresso.Logger
+import com.iron.espresso.ValidationInputText
+import com.iron.espresso.base.BaseViewModel
+import com.iron.espresso.data.model.*
+import com.iron.espresso.domain.repo.StudyRepository
+import com.iron.espresso.ext.Event
+import com.iron.espresso.ext.networkSchedulers
+import com.iron.espresso.ext.plusAssign
+import com.iron.espresso.ext.toErrorResponse
+import com.iron.espresso.model.api.ModifyStudyRequest
+import retrofit2.HttpException
+
+class ModifyStudyViewModel @ViewModelInject constructor(private val studyRepository: StudyRepository) :
+    BaseViewModel() {
+
+    private val _localItem = MutableLiveData<LocalItem>()
+    val localItem: LiveData<LocalItem>
+        get() = _localItem
+
+    private val _emptyCheckMessage = MutableLiveData<Event<ValidationInputText>>()
+    val emptyCheckMessage: LiveData<Event<ValidationInputText>>
+        get() = _emptyCheckMessage
+
+    private val _success = MutableLiveData<Event<Unit>>()
+    val success: LiveData<Event<Unit>>
+        get() = _success
+
+    val image = MutableLiveData<Uri>()
+
+    fun setStudyImage(image: Uri) {
+        this.image.value = image
+    }
+
+    fun addItems(localItem: LocalItem?) {
+        if (localItem != null) {
+            _localItem.value = LocalItem(
+                lat = localItem.lat,
+                lng = localItem.lng,
+                sido = localItem.sido,
+                sigungu = localItem.sigungu,
+                addressName = localItem.addressName,
+                placeName = localItem.placeName,
+                locationDetail = localItem.locationDetail
+            )
+        }
+    }
+
+    private fun emptyCheck(
+        modifyStudyItem: ModifyStudyItem
+    ): ValidationInputText {
+        return when {
+            modifyStudyItem.title.isEmpty() -> ValidationInputText.EMPTY_TITLE
+            modifyStudyItem.introduce.isEmpty() -> ValidationInputText.EMPTY_INTRODUCE
+            modifyStudyItem.progress.isEmpty() -> ValidationInputText.EMPTY_PROGRESS
+            modifyStudyItem.studyTime.isEmpty() -> ValidationInputText.EMPTY_TIME
+            else -> ValidationInputText.REGISTER_STUDY
+        }
+    }
+
+    private fun duplicateItemCheck(title: String, modifyTitle: String): String {
+        return if (title == modifyTitle) {
+            ""
+        } else {
+            modifyTitle
+        }
```

Comment:
```suggestion
        return if (title == modifyTitle) "" else modifyTitle
```
이런 경우(블록 내의 계산이 없는경우) java 삼항 연산자와 비슷하게 이렇게 작성하는 경우가 많음
`title == modifyTitle ? "" : modifyTitle`

Reply:
수정하였습니다
b564064